mailserver/mailserver.go

```diff
@@ -212,23 +224,32 @@ func (s *WMailServer) exceedsPeerRequests(peer []byte) bool {
 
 // processRequest processes the current request and re-sends all stored messages
 // accomplishing lower and upper limits.
-func (s *WMailServer) processRequest(peer *whisper.Peer, lower, upper uint32, bloom []byte) []*whisper.Envelope {
-	ret := make([]*whisper.Envelope, 0)
-	var err error
-	var zero common.Hash
-	kl := NewDbKey(lower, zero)
-	ku := NewDbKey(upper+1, zero) // LevelDB is exclusive, while the Whisper API is inclusive
-	i := s.db.NewIterator(&util.Range{Start: kl.raw, Limit: ku.raw}, nil)
-	defer i.Release()
-
+func (s *WMailServer) processRequest(peer *whisper.Peer, lower, upper uint32, bloom []byte, limit uint32, cursor cursorType) ([]*whisper.Envelope, common.Hash, cursorType) {
 	var (
-		sentEnvelopes     int64
+		sentEnvelopes     uint32
 		sentEnvelopesSize int64
+		nextPageCursor    cursorType
+		lastEnvelopeHash  common.Hash
+		err               error
+		zero              common.Hash
+		ku                []byte
```

Comment:
Maybe it would be cleaner to declare `kl` here as well? No strong opinion on it though, just a thought.